testing.go

```diff
@@ -717,20 +717,21 @@ WAIT:
 
 // NOTE: This is exposed for middleware testing purposes and is not a stable API
 type MakeClusterOpts struct {
-	Peers           int
-	Bootstrap       bool
-	Conf            *Config
-	ConfigStoreFSM  bool
-	MakeFSMFunc     func() FSM
-	LongstopTimeout time.Duration
-	MonotonicLogs   bool
+	Peers              int
+	Bootstrap          bool
+	Conf               *Config
+	ConfigStoreFSM     bool
+	MakeFSMFunc        func() FSM
+	LongstopTimeout    time.Duration
+	MonotonicLogs      bool
+	CommitTrackingLogs bool
 }
 
 // makeCluster will return a cluster with the given config and number of peers.
 // If bootstrap is true, the servers will know about each other before starting,
 // otherwise their transports will be wired up but they won't yet have configured
 // each other.
-func makeCluster(t *testing.T, opts *MakeClusterOpts) *cluster {
+func makeCluster(t *testing.T, opts *MakeClusterOpts) (*cluster, error) {
```

Comment:
AFAICT this was added to support 1 test that asserts a specific error is returned. Since this causes tons of other test churn I have a slight preference for reverting the error return here and special casing the 1 test that needs to check the error return of NewRaft. Not a blocker though.